src/shared/ui/components/InstructionsTable.vue

```diff
@@ -45,36 +44,50 @@ function getInstructionPayloadEntity(item: Instruction) {
   return Object.entries(item.payload)[0][0];
 }
 
-const instructionsTable = useTable(http.fetchInstructions);
-
-async function fetchInstructions() {
-  try {
-    if (isOnAccountPage.value) {
-      await instructionsTable.fetch({
-        ...objectOmit(listState, shouldShowKind.value ? ['kind'] : []),
-        transaction_status: listState.status,
-      });
-    } else
-      await instructionsTable.fetch({
-        ...objectOmit(listState, shouldShowKind.value ? ['kind'] : []),
-      });
-  } catch (e) {
-    handleUnknownError(e);
-  }
-}
-
 const isOnAccountPage = computed(() => props.filterBy.kind === 'authority');
+const shouldShowKind = computed(() => listState.kind === 'All');
 
 const listState = reactive({
-  status: null,
-  authority: computed(() => props.filterBy.kind === 'authority' && props.filterBy.value.toString()),
-  transaction_hash: computed(() => props.filterBy.kind === 'transaction' && props.filterBy.value),
+  transaction_status: null,
+  authority: computed(() => (props.filterBy.kind === 'authority' ? props.filterBy.value.toString() : '')),
+  transaction_hash: computed(() => (props.filterBy.kind === 'transaction' ? props.filterBy.value : '')),
   kind: 'All' as ftm.TabInstructions,
+  page: 1,
+  per_page: 10,
 });
 
-const shouldShowKind = computed(() => listState.kind === 'All');
+async function fetchInstructions(params: Reactive<typeof listState>) {
+  if (isOnAccountPage.value) {
+    return await http.fetchInstructions({
+      ...objectOmit(params, shouldShowKind.value ? ['kind'] : []),
+      transaction_status: params.transaction_status ?? undefined,
+    });
+  } else
+    return await http.fetchInstructions({
+      ...objectOmit(params, shouldShowKind.value ? ['kind', 'transaction_status'] : ['transaction_status']),
+    });
+}
 
-watch(listState, fetchInstructions, { immediate: true });
+watch(
+  () => [listState.kind, listState.transaction_status, listState.per_page],
```

Comment:
This watch will trigger even if the actual values in the array aren't changed (because the array object will be new each time this function is called).

Make three separate watch sources instead:
```suggestion
  [() => listState.kind, () => listState.transaction_status, () => listState.per_page],
```